File: 1. Modelling challenge/Scoring/scoring_script.py
#  you need pandas and numpy for this script to run. 
#  from command line:
#  pip install pandas
#  pip install numpy

import pandas as pd
import numpy as np

#  CHANGE SEPARATOR AND FILE PATHS ACCORDINGLY #

actuals = pd.read_csv("actuals.csv", sep=',')
preds = pd.read_csv("predictions.csv", sep=',')
data = pd.merge(actuals,preds, on='admin_L3_code')
R2 = 1-sum((data.actuals-data.predictions)**2)/sum((data.actuals-np.mean(data.actuals))**2)
print("R squared: ", R2)

File: 1. Modelling challenge/Scoring/scoring_script.R

#Adjust the file path & beware of the correct separator (your Excel might store csv with ',' or ';')

#getwd()
setwd("C:/Users/JannisV/Desktop/Hackathon/1. Modelling challenge/Scoring")
actuals <- read.csv(file="./actuals.csv", header=TRUE, sep=",")
#View(actuals)
preds <- read.csv(file="./predictions.csv", header=TRUE, sep=",")
#View(preds)

data <- merge(x = actuals, y = preds, by.x=c("typhoon_name","admin_L3_code"), by.y=c("typhoon_name","admin_L3_code"))
#View(data)

R2 <- 1 - (sum((data$actuals-data$predictions)^2)/sum((data$actuals-mean(data$actuals))^2))



